### PR TITLE
feat: update combo-box to account for 'create new' autocomplete use cases

### PR DESCRIPTION
**Pull request by jake-bassett**
## Description
This adds the ability to "Create new" value using combo-box autocomplete.

### CODE REVIEW

File: projects/components/src/combo-box/combo-box.component.ts [hunk 1]
```diff
@@ -144,30 +165,49 @@ export class ComboBoxComponent<TValue = string> implements AfterViewInit, OnChan
...
+      text: `Create "${text}"`,
+      tooltip: text,
+      icon: IconType.AddCircleOutline
+    };
+  }
+
   private setFilteredOptions(text?: string): void {
-    this.filteredOptions = (this.options ?? []).filter(option => text === undefined || option.text.includes(text));
+    this.filteredOptions = (this.options ?? []).filter(
+      option => text === undefined || option.text.toLowerCase().includes(text.toLowerCase())
```
Comment (anandtiwary) [inline projects/components/src/combo-box/combo-box.component.ts:208]: if text is undefined then we don't need to call filter.

File: projects/components/src/combo-box/combo-box.component.ts [hunk 2]
```diff
@@ -144,30 +165,49 @@ export class ComboBoxComponent<TValue = string> implements AfterViewInit, OnChan
...
+    return {
+      text: `Create "${text}"`,
+      tooltip: text,
+      icon: IconType.AddCircleOutline
+    };
+  }
+
   private setFilteredOptions(text?: string): void {
-    this.filteredOptions = (this.options ?? []).filter(option => text === undefined || option.text.includes(text));
+    this.filteredOptions = (this.options ?? []).filter(
```
Comment (anandtiwary) [inline projects/components/src/combo-box/combo-box.component.ts:207]: Can we call it `searchText`?

File: projects/components/src/combo-box/combo-box.component.ts [hunk 3]
```diff
@@ -210,25 +252,63 @@ export class ComboBoxComponent<TValue = string> implements AfterViewInit, OnChan
...
+        this.highlightedOptionIndex = 0;
+      } else if (this.highlightedOptionIndex >= this.filteredOptions.length - 1) {
+        this.highlightedOptionIndex = -1;
+      }
+    } else {
       this.highlightedOptionIndex = ++this.highlightedOptionIndex % this.filteredOptions.length;
-      event.preventDefault(); // Prevent tabbing to next control on page
     }
+
+    this.scrollIntoViewService.scrollIntoView(this.optionElements.get(this.highlightedOptionIndex)?.nativeElement);
```
Comment (anandtiwary) [inline projects/components/src/combo-box/combo-box.component.ts:282]: Nice. Is it this easy to scroll and show a particular element in the view? I can use it at so many other places now.

**Review by anandtiwary — APPROVED** [review]

File: projects/components/src/combo-box/combo-box.component.ts (hunk 3 above)
Comment (jake-bassett) [reply projects/components/src/combo-box/combo-box.component.ts:282]: Yep! Just this easy.

**Review by anandtiwary — APPROVED** [review]

**Pull request merged**